runtime/src/commonMain/kotlin/com/squareup/sqldelight/db/SqlDriver.kt

```diff
@@ -90,3 +90,41 @@ interface SqlDriver : Closeable {
     fun migrate(driver: SqlDriver, oldVersion: Int, newVersion: Int)
   }
 }
+
+/**
+ * Represents a block of code [block] that should be executed during a migration after the migration
+ * has finished migrating to [afterVersion].
+ */
+class AfterVersion(
+  internal val afterVersion: Int,
+  internal val block: () -> Unit
+)
+
+/**
+ * Run [SqlDriver.Schema.migrate] normally but execute [callbacks] during the migration whenever
+ * the it finished upgrading to a version specified by [AfterVersion.afterVersion].
+ */
+fun SqlDriver.Schema.migrateWithCallbacks(
+  driver: SqlDriver,
+  oldVersion: Int,
+  newVersion: Int,
+  vararg callbacks: AfterVersion
+) {
+  var lastVersion = oldVersion
+
+  // For each callback within the oldVersion..newVersion range, alternate between migrating
+  // the schema and invoking each callback.
+  callbacks.filter { it.afterVersion in (oldVersion + 1)..newVersion }
+    .sortedBy { it.afterVersion }
+    .forEach { callback ->
+      migrate(driver, oldVersion = lastVersion, newVersion = callback.afterVersion + 1)
```

Comment:
The +1 here doesn't seem right. If you're on version 1, are migrating to 5, and want to run after version 3 wouldn't we want to call `migrate(old = 1, new = 3); callback(); migrate(old = 3, new = 5)`?

Reply:
if you want to run after version 3, that means you want to run when the database is on version 4

Reply:
if you have `5.sqm` as your final migration, the database will be on version 6 after migrating is complete

Reply:
which is also why I'm going with `AfterVersion` since it matches how we're modeling things in sqldelight.

3.sqm
afterVersion(3) { ... } (you are on version 3 here)
4.sqm
5.sqm (you are on version 6 after running migration)

Reply:
I see. It would be great to copy that little example into the docs you added to make it unambiguously clear.

Reply:
yep, can do